classes/user/form/PublicProfileForm.php

```diff
@@ -160,6 +161,25 @@ public function fetch($request, $template = null, $display = false)
             'publicSiteFilesPath' => $publicFileManager->getSiteFilesPath(),
         ]);
 
+        // TODO: Check how to handle ORCID at site-level
```

Comment:
I agree. I think having a checkbox for the site-level will help simplify the check. I think this will mean a few more custom validation changes to the Vue forms, but that won't be a problem. The site-level vs. context-level checking is complex enough we would need it in any case.